src/tests/unit/auto/plugin/mock_infer_request.hpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+#include <gmock/gmock.h>
+#include "ie_icore.hpp"
+#include "plugin.hpp"
+#include <iostream>
+
+using namespace MockMultiDevicePlugin;
+namespace MockMultiDevice {
+
+struct mockRequestExecutor : public IE::ITaskExecutor {
+public:
```

Comment:
I think this is same as ImmediateExecutor?

Reply:
Yes, will replace it with ImmediateExecutor. Thanks.